Content.Server/GameTicking/Rules/RevolutionaryRuleSystem.cs

```diff
@@ -0,0 +1,336 @@
+                var wrappedMessage = Loc.GetString("chat-manager-server-wrap-message", ("message", message));
+                _chatManager.ChatMessageToOne(Shared.Chat.ChatChannel.Server, message, wrappedMessage, default, false, mind.Session.ConnectedClient, Color.MediumBlue);
+            }
+        }
+        AssignCommandStaff(cantBeAntag);
+    }
+
+    //Assigns command staff based on their weight (real). I THINK all head weight is above 10 so uh let me know if not.
+    //I wanted to do sec as well but I'm not sure if they have a specific weight and they have glasses so if they get converted it's their own fault.
+    private void AssignCommandStaff(List<IPlayerSession> players)
+    {
+        foreach (var player in players)
+        {
+            var mind = player.GetMind();
+            if (mind != null && mind.CurrentJob != null)
+            {
+                if (mind.OwnedEntity != null && mind.CurrentJob.Prototype.Weight >= 10)
+                {
+                    AddComp<HeadComponent>(mind.OwnedEntity.Value);
+                }
+            }
+        }
+    }
+
+
+    private void OnMobStateChanged(MobStateChangedEvent ev)
+    {
+        if (ev.NewMobState == MobState.Dead)
+            CheckRevsLose(ev.Target);
+    }
+
+    private void CheckRevsLose(EntityUid target)
+    {
+        var stunTime = TimeSpan.FromSeconds(4);
+        var query = EntityQueryEnumerator<RevolutionaryRuleComponent, GameRuleComponent>();
+        while (query.MoveNext(out var uid, out var revs, out var gameRule))
+        {
+            if (!GameTicker.IsGameRuleAdded(uid, gameRule))
+                continue;
+
+            var headRevs = EntityQuery<HeadRevolutionaryComponent, MobStateComponent>(true);
+            var aliveHeadRevs = headRevs
+                .Any(ent => ent.Item2.CurrentState != MobState.Dead && ent.Item1.Running);
+            //If no Head Revs are alive all normal Revs will lose their Rev status and rejoin Nanotrasen
+            if (!aliveHeadRevs)
+            {
+                revs.RevsLost = true;
+                var allPlayers = _playerSystem.ServerSessions.ToList();
+                foreach (var rev in allPlayers)
+                {
+                    var mind = rev.GetMind();
+                    if (mind != null)
+                        if (HasComp<RevolutionaryComponent>(mind.OwnedEntity) && !HasComp<HeadRevolutionaryComponent>(mind.OwnedEntity))
+                        {
+                            _npcFaction.AddFaction(mind.OwnedEntity.Value, "NanoTrasen");
+                            _npcFaction.RemoveFaction(mind.OwnedEntity.Value, "Revolutionary");
+                            _sharedStun.TryParalyze(mind.OwnedEntity.Value, stunTime, true);
+                            RemComp<RevolutionaryComponent>(mind.OwnedEntity.Value);
+                        }
+                }
+            }
+            CheckRevsWin();
+        }
+    }
```

Comment:
1. Please fix your formatting, e.g. stuntime is completely unused, you have a braceless if in an ugly spot, etc
2. Please don't copy-paste code.